src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -1364,6 +1364,18 @@ void StorageReplicatedMergeTree::checkParts(bool skip_sanity_checks)
 
     paranoidCheckForCoveredPartsInZooKeeperOnStart(expected_parts_vec, parts_to_fetch);
 
+    ActiveDataPartSet set_of_empty_unexpected_parts(format_version);
+    for (const auto & part : parts)
+    {
+        if (part->rows_count || part->getState() != MergeTreeDataPartState::Active || expected_parts.contains(part->name))
```

Comment:
This is accessing part->getState() and that needs lockParts(), right?

Reply:
It's a good question. Usually we need the lock for this, but checkParts() is called during table initialization when there shouldn't be other threads that can change state of parts